src/panels/config/core/ha-config-url-form.ts

```diff
@@ -0,0 +1,162 @@
+                      <p>
+                        ${this.hass.localize(
+                          "ui.panel.config.core.section.core.core_config.edit_requires_storage"
+                        )}
+                      </p>
+                    `
+                  : ""}
+
+                <div class="row">
+                  <div class="flex">
+                    ${this.hass.localize(
+                      "ui.panel.config.core.section.core.core_config.external_url"
+                    )}
+                  </div>
+
+                  <paper-input
+                    class="flex"
+                    .label=${this.hass.localize(
+                      "ui.panel.config.core.section.core.core_config.external_url"
+                    )}
+                    name="external_url"
+                    type="url"
+                    .disabled=${disabled}
+                    .value=${this._externalUrlValue}
+                    @value-changed=${this._handleChange}
+                  >
+                  </paper-input>
+                </div>
+
+                <div class="row">
+                  <div class="flex">
+                    ${this.hass.localize(
+                      "ui.panel.config.core.section.core.core_config.internal_url"
+                    )}
+                  </div>
+                  <paper-input
+                    class="flex"
+                    .label=${this.hass.localize(
+                      "ui.panel.config.core.section.core.core_config.internal_url"
+                    )}
+                    name="internal_url"
+                    type="url"
+                    .disabled=${disabled}
+                    .value=${this._internalUrlValue}
+                    @value-changed=${this._handleChange}
+                  >
+                  </paper-input>
+                </div>
+              </div>
+              <div class="card-actions">
+                <mwc-button @click=${this._save} .disabled=${disabled}>
+                  ${this.hass.localize(
+                    "ui.panel.config.core.section.core.core_config.save_button"
+                  )}
+                </mwc-button>
+              </div>
+            </ha-card>
+          `
+        : ""}
+    `;
+  }
+
+  private get _internalUrlValue() {
+    return this._internal_url !== undefined
+      ? this._internal_url
+      : this.hass.config.internal_url;
+  }
+
+  private get _externalUrlValue() {
+    return this._external_url !== undefined
+      ? this._external_url
+      : this.hass.config.external_url;
+  }
+
+  private _handleChange(ev: PolymerChangedEvent<string>) {
+    const target = ev.currentTarget as PaperInputElement;
+    this[`_${target.name}`] = target.value;
+  }
+
+  private async _save() {
+    this._working = true;
+    try {
+      await saveCoreConfig(this.hass, {
+        external_url: this._external_url || null,
+        internal_url: this._internal_url || null,
+      });
+    } catch (err) {
+      alert("Invalid URLs");
```

Comment:
Don't show an alert but save the error you get from the backend in `this._error` and print that at the top of the form in the `error-color`. 
```ts
${this._error ? html`<div class="error">${this._error}</div>` : ""}
```

Reply:
And reset the error when you start saving.

Reply:
Done!